[Level 2] 엄서원 학습 PR 제출합니다

EomSeoWon opened the pull request.

EomSeoWon committed:
Update README.md

EomSeoWon committed:
[Level 2]엄서원 학습 PR 제출합니다.

EomSeoWon committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	level2/README.md

EomSeoWon committed:
[Level 2]엄서원 학습 PR 제출합니다.

KoSeonJe reviewed (commented):
README는 수정해서는 안됩니다. 원상태로 복구해주시고,
본인이 구현한 프로젝트 파일을 제출해주세요.

KoSeonJe reviewed (commented):
보통 main에는 구현 로직이 있어서는 안됩니다. 실행하고자 하는 객체의 행위를 간단히 호출해야하는 것이죠.

즉, 첫 번째로 문자열 계산기라는 객체를 만들고, 문자열 계산기에게 계산을 해라. 라고 단순히 호출해야 한다는 것입니다. 그리고 문자열 계산기는 사용자에게 입력을 받는 행위를 하거나, 사용자에게 콘솔을 통해 결과를 화면에 출력해주거나, 문자열을 분리하거나, 계산하거나 이런 다양한 책임을 가진 객체를 또 하위에 만들어서 서로 상호작용하게 만들어야 합니다.

현재 객체 분리가 부족한 것 같습니다. 정말 처음부터 하기 어렵고 힘들지만 많이 고민하면 고민할 수록 얻어지는 것이 있으니 객체를 분리해보는 연습을 하면 좋겠습니다!

또한, 코드 컨벤션이 지켜지지 않고 있습니다. 신경써주세요. 한글로 치면 맞춤법, 영어로는 문법과 같은 것입니다.
관련 링크입니다
https://sihyung92.oopy.io/af26a1f6-b327-45a6-a72b-c6fcb754e219

EomSeoWon committed:
[Level 2]엄서원 학습 PR 제출합니다.

EomSeoWon commented:
package Homework2;

import java.util.*;
import java.util.Scanner;
import java.util.regex.Pattern;

public class Level2 {
public static void main(String[] args) {
Scanner sc = new Scanner(System.in);
System.out.println("덧셈할 문자열을 입력해 주세요.");
Calculator calculator = new Calculator();
calculator.inputNumbers();
}
}

class Calculator {

    public Calculator() {
    }

    public void inputNumbers() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("덧셈할 문자열을 입력해 주세요.");

        String input = scanner.nextLine();
        calculateSum(input);

        String input = sc.nextLine();
        scanner.close();
    }

    public void calculateSum(String input) {
        
        if (input.isEmpty()) {
            System.out.println("결과: 0");
            return;
        }

        String parts[] = exchange(input);
        
        String[] parts = splitInput(input);
        int sum = 0;

        try {
            for (String part : parts) {
                
                sum += Integer.parseInt(part.trim());
            }

            System.out.println("결과: " + sum);
        } catch (IllegalArgumentException e) {
        } catch (NumberFormatException e) {
            System.out.println("숫자로 변환할 수 없는 값이 포함되어 있습니다.");
        }

        sc.close();
    }

    public static String[] exchange(String input) {

    public String[] splitInput(String input) {
        
        if (input.startsWith("//")) {
            int index = input.indexOf('\n');
            String c = input.substring(2,index);
            if (index == -1) {
                System.out.println("잘못된 사용자 정의 구분자 형식입니다.");
                return new String[0];
            }

           
            String delimiter = input.substring(2, index);

            String numbers = input.substring(index + 1);

            String num = input.substring(index + 1);
           
            String result = "[" + c + "]";
            String[] parts1 = num.split(result);
            return parts1;
            String escapedDelimiter = Pattern.quote(delimiter);
            return numbers.split(escapedDelimiter);
        }

        String[] parts2 = input.split("[,:]");
        return parts2;
        return input.split("[,:]");
    }
}

EomSeoWon committed:
[Level 2]-2 엄서원 학습 PR 제출합니다.

KoSeonJe reviewed (commented):
README 수정하지말고 자신이 진행한 프로젝트를 제출해주면 좋을 것 같습니다!
